cluster/manifests/metrics-server/metrics-server-vpa.yaml

```diff
@@ -10,4 +10,11 @@ spec:
       application: metrics-server
   updatePolicy:
     updateMode: "Auto"
+  {{if index .ConfigItems "prometheus_mem"}}
+  resourcePolicy:
+    containerPolicies:
+    - containerName: metrics-server
+      maxAllowed:
+        memory: {{.ConfigItems.prometheus_mem}}
```

Comment:
Wouldn't it be better to set it directly:

`memory: {{if index .ConfigItems "prometheus_mem"}}{{.ConfigItems.prometheus_mem}}{{else}}4000Mi{{end}}`

Reply:
same for **prometheus-vpa.yaml**

Reply:
Beacause the `MaxResourcePolicy` sets the ceiling for how much the VPA can scale a container **up**, if we setup a default value e.g `4000Mi` than the VPA would never scale beyond that.i.e

```
requests <.....vpa scale up range.....> MaxResourcePolicy
```
I'm just worried that this could hurt us in large clusters ( large node sizes and high no. of nodes). 

I guess the question is, _"what is more likely?"_ The case where the VPA scales prometheus up so much that it becomes unschedulable  or that the `MaxResourcePolicy` would be set too low and the VPA won't be able to scale up `prometheus` enough? 

What do you think?

Reply:
Either way it won't be any different from the current behavior

Reply:
Sounds good. I'll add a default value

Reply:
I'd set the default in `config-defaults.yaml` and just use the config item in here (and set it to a much larger value, e.g. `14Gi`).